### ISSUE TITLE
cannot install to paths in different partition

### ISSUE THREAD

**Issue by christianrauch** [flathub/com.heroicgameslauncher.hgl#130]
When changing the path in `Select Install Path` to something pointing to another partition, e.g. `/media/user/DATA1/Heroic/`, it gets replaced by `/run/user/1000/doc/5e5c93a/Heroic` with the yellow hint `Warning: path might not be writable.`. However, both paths (the original path and `/run/user/1000/doc/5e5c93a/Heroic`) are writeable by the user when using a terminal or file manager.

**Comment by Gamertech** [comment]
Flatpaks are sandboxed and need to be given permission to access directories via Flatseal or CLI.

If they don't have permission, they go through the system portal via a temporary path which is the /run/user/1000.

Download the Flatseal flatpak and give Heroic file permissions to access the directory and reset the install path.

**Comment by Gamertech** [comment]
If you use one of the many common storage paths that are already permitted, then it does work out of the box. The Heroic flatpak even permits /mnt by default, which is best practice for mounting additional storage.

The flatpak works fine by default. It's if you want to customise things that you need to change configs, like any other install method, and in this case only if you want to set a custom storage location outside the many already permitted areas.

Flatpaks offer sandboxed security that other methods don't. eg. Discord's practice of auto-caching and running malware can't happen with the flatpak, Steam on Linux swipes stored wifi passwords and auth keys every launch, but is blocked by flatpak.

You can edit permissions with the terminal, but Flatseal makes it simple. KDE Plasma also has their own flatpak UI built into system settings. Just add the path you want to the whitelist, then you can set an absolute path and forget about it forever more. Or you can use /mnt or ~/Games/Heroic and it just works.

**Comment by christianrauch** [comment]
Let me rephrase the issue: This is not really about configuring the paths that the flatpak can access natively, but about the portal not working correctly. When I use the portal to select a path under `Select Install Path`, the returned path (`/run/user/1000/doc/5e5c93a/Heroic`) should be writeable. But in the case above it is not. Setting the file system access via Flatseat or `--filesystem` is really just a workaround around the portal. You also shouldn't need `--filesystem=/mnt` that is currently in the flatpak definition.

Or am I overlooking something here? Is the portal supposed to work that way or do you also have to configure the portal in some way manually? I thought managing the access to the filesystem and other functionality securely is exactly what the portals were created for.

**Comment by imLinguin** [comment]
It's Heroic that returns an error that "the path isn't writeable" for all `/run/user/1000/doc/....`. The reasoning behind this is simple, the path is temporary, Heroic needs static path in order to be able, well... to even launch the downloaded game after system reboot. 

At the moment we recommend users to add necessary permissions to their locations on their own. I'm fine with adding more default paths to the manifest to make this work out of the box.
